crates/uv-build-frontend/src/error.rs

```diff
@@ -131,6 +138,12 @@ pub struct MissingHeaderCause {
 impl Display for MissingHeaderCause {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         match &self.missing_library {
+            MissingLibrary::NonNormalizedSetupCfgKey => {
+                write!(
+                    f,
+                    "setuptools deprecated support for dash-separated and uppercased keys in v78.0.0. Consider updating the relevant `setup.cfg`, or use `tool.uv.build-constraint-dependencies` to pin `setuptools<78`:\n```toml\n[tool.uv]\nbuild-constraint-dependencies = [\"setuptools<78\"]\n```"
```

Comment:
```suggestion
                    "setuptools removed support for dash-separated and uppercased keys in v78.0.0. Consider updating the relevant `setup.cfg`, or use `tool.uv.build-constraint-dependencies` to pin `setuptools<78`:\n```toml\n[tool.uv]\nbuild-constraint-dependencies = [\"setuptools<78\"]\n```"
```

Deprecation implies warning not erroring to me.